libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -425,6 +447,17 @@ private static Stream<InstrumentationService.InstrumentationInfo> pathChecks() {
         });
     }
 
+    private static void ensureClassesSensitiveToVerificationAreInitialized() {
+        var classesToInitialize = Set.of("sun.net.www.protocol.http.HttpURLConnection");
```

Comment:
This needs a comment. As I understand it, this is because verification fails if a certain load order occurs? This still seems like a JDK bug, so it should at least be documented this is a workaround for that.

Reply:
I don't think it is; the verification fails/has issues depending on the order in which we _transform_ and _verify_ things; I think we are at least in an unforeseen (if not unsupported) scenario here: we are loading a class, and while loading it we try to verify it, which in order leads to more classes loading, which _could_ turn into verification...
Same reason why we can't have this always on, not even in snapshot builds. It's a good instrument if we need to get more info in case of a VerifyError, but it's not without side effects.

There is a comment already at the call site, I'll move that here and expand.

Reply:
LogThe reason a class is being transformed (normally) is it is about to be reloaded. In this case, there should be no difference since the class will be reloaded anyways.

I wonder if we need to be smarter about the order we pass the classes to be retransformed. Normally the jdk would load these in the appropriate order based on hierarchy? Or perhaps calling retransform one at a time is actually the safer approach?

Reply:
> The reason a class is being transformed (normally) is it is about to be reloaded. In this case, there should be no difference since the class will be reloaded anyways.

++ loaded or reloaded.
It might be that all of this only matters for retransform in conjunction with verify; in that case ordering how we retransform classes should solve the issue.